lib/puppet/pops/types/types.rb

```diff
@@ -203,6 +230,29 @@ def hash
         31 * @type.hash
       end
 
+      def iterable?
+        case @type
+        when PEnumType
+          true
+        when PIntegerType
+          @type.enumerable?
+        else
+          false
+        end
+      end
+
+      def iterable_type
+        # The types PIntegerType and PEnumType are Iterable
+        case @type
+        when PEnumType
+          @type.each
+        when PIntegerType
+          @type.enumerable? ? @type.each : nil
```

Comment:
No. `iterable?` and `enumerable?` has different meaning (type versus instances of the type). I did add specific method documentation to explain what they do.